docs/kube-api-custom-ocp-versions.md

```diff
@@ -0,0 +1,94 @@
+* As a reference to the ClusterImageSet in ```spec.provisioning.imageSetRef``` property.
+* Explicitly as a URL in ```spec.provisioning.releaseImage``` property.
+
+An example of a ClusterImageSet:
+```
+apiVersion: hive.openshift.io/v1
+kind: ClusterImageSet
+metadata:
+  name: openshift-v4.7.0
+spec:
+  releaseImage: quay.io/openshift-release-dev/ocp-release:4.7.0-x86_64
+```
+
+### [OPENSHIFT_VERSIONS](https://github.com/openshift/assisted-service/blob/master/Makefile#L36)
+
+The OPENSHIFT_VERSIONS environment variable is a map of OpenShift version (y-stream) to related properties as release version/image and RHCOS version/image: see [default_ocp_versions.json](https://github.com/openshift/assisted-service/blob/master/default_ocp_versions.json)
+
+Defining this var prior to starting the service would override all versions supported by the service. I.e. by default, [default_ocp_versions.json](https://github.com/openshift/assisted-service/blob/master/default_ocp_versions.json) is used for setting the supported versions. Defining OPENSHIFT_VERSIONS with a similar json format should override these.
+
+## Flow
+
+The flow of adding a new version is a follows:
+* OPENSHIFT_VERSIONS is defined with a map of "x.y" -> ```rhcos_image``` and ```rhcos_version```.
+* Upon starting the service, the relevant host [boot-files](https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L16) are uploaded to S3/File storage.
+* ClusterImageSet with a new ```releaseImage``` is deployed to Hive cluster.
+* The name of the ClusterImageSet is referenced in the ClusterDeployment (```spec.provisioning.imageSetRef``` property).
+* Upon deploying the ClusterDeployment:
+  * ClusterImageSet is fetched according to ```imageSetRef```.
+  * ```releaseImage``` is read from the ClusterImageSet.
+  * The ```release_version``` is extracted from the release image (using ```oc adm release info```) - in major.minor.patch format.
+  * According to the major.minor as key, the relevant entry is updated in the [versions cache](https://github.com/openshift/assisted-service/blob/master/internal/versions/versions.go). I.e. ```release_version``` and ```display_name``` properties.
+* Finally, a new cluster is created with the newly added [openshift_version](https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L4106). 
+
+
+## Usage
+
+### Set OPENSHIFT_VERSIONS
+
+OPENSHIFT_VERSIONS should be set with ```rhcos_image``` and ```rhcos_version``` in the following format:
+```
+{
+  "major.minor": {
+    "rhcos_image":"<image URL>",
+    "rhcos_version":"<version>"
+  }
+}
+```
+
+E.g.
+```
+export OPENSHIFT_VERSIONS='{"4.7":{"rhcos_image":"https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.7/4.7.0/rhcos-4.7.0-x86_64-live.x86_64.iso","rhcos_version":"47.83.202102090044-0"}}'
+```
+
+### Deploy ClusterImageSet
+
+Deploy a ClusterImageSet with the requested release image.
+
+E.g.
+```
+apiVersion: hive.openshift.io/v1
```

Comment:
Maybe have the yamls under also in `docs/crds`

Reply:
+1